### PR TITLE
docs: Add documentation about access logs PoC

### PR DESCRIPTION
**Pull request by chrkl**
## Description

Changes proposed in this pull request (what was done and why):

- Add documentation about access logs PoC

Changes refer to particular issues, PRs or documents:

- #821 

## Traceability
- [ ] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [ ] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] Adjusted the documentation if the change is user-facing.
- [ ] The feature is unit-tested
- [ ] The feature is e2e-tested

### CODE REVIEW

File: docs/contributor/pocs/otlp-logs.md [hunk 1]
```diff
@@ -167,3 +167,161 @@ During the evaluation, the following potential problems and risks have been iden
...
+  },
+  "sort": [
+    1710427383130026000
+  ]
+}
+```
+
+The configured labels from the extension provider configuration result in log attributes of the OpenTelemetry log records. The log gateway enriches the log records with resource attributes to fulfill the semantic conventions. The source pod is identified by the k8sattributes processor using the client IP address.
+
+The OpenTelemetry log specification contains fields for a trceId and spanId. The extension provider does not allow to fill these fields but assign them to a log attribute. This limitation can be corrected by a transform processor in the log gateway. An [issue](https://github.com/istio/istio/issues/49911) has been created for the Istio project.
```
Comment (rakesh-garimella) [inline docs/contributor/pocs/otlp-logs.md:323]:
```suggestion
The OpenTelemetry log specification contains fields for a traceId and spanId. The extension provider does not allow to fill these fields but assign them to a log attribute. This limitation can be corrected by a transform processor in the log gateway. An [issue](https://github.com/istio/istio/issues/49911) has been created for the Istio project.
```

**Review by rakesh-garimella — APPROVED** [review]

File: docs/contributor/pocs/otlp-logs.md [hunk 2]
```diff
@@ -167,3 +167,161 @@ During the evaluation, the following potential problems and risks have been iden
...
+
+The OpenTelemetry log specification contains fields for a traceId and spanId. The extension provider does not allow to fill these fields but assign them to a log attribute. This limitation can be corrected by a transform processor in the log gateway. An [issue](https://github.com/istio/istio/issues/49911) has been created for the Istio project.
+
+The OpenTelemetry [Semantic Conventions](https://opentelemetry.io/docs/specs/semconv/) do not explicitly cover access logs yet. The shown example pick up attribute names from related areas like HTTP and URL where it made sense.
+
+We can conclude that the EnvoyOpenTelemetryLogProvider allows to emit access logs to an OTLP endpoint with similar properties that the currently used JSON based format in combination with Fluent Bit provides. Emitting access logs using the OTLP protocol has the advantage that regular Istio sidecar logs can be collected in the same way as regular application logs and are more easily available for troubleshooting if necessary.
```
Comment (NHingerl) [inline docs/contributor/pocs/otlp-logs.md:327]:
```suggestion
We conclude that the EnvoyOpenTelemetryLogProvider supports emitting access logs to an OTLP endpoint with similar properties that the currently used JSON-based format provides in combination with Fluent Bit. Emitting access logs using the OTLP protocol has the advantage that regular Istio sidecar logs can be collected in the same way as regular application logs and are more easily available for troubleshooting if necessary.
```

File: docs/contributor/pocs/otlp-logs.md [hunk 3]
```diff
@@ -167,3 +167,161 @@ During the evaluation, the following potential problems and risks have been iden
...
+
+Istio comes with extension providers to configure its telemetry. The [EnvoyOpenTelemetryLogProvider](https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig-ExtensionProvider-EnvoyOpenTelemetryLogProvider) allows to send access logs to an OTLP service. This part of the PoC should compare this extension provider in combination with the OpenTelemetry Collector to the [EnvoyFileAccessLogProvider](https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig-ExtensionProvider-EnvoyFileAccessLogProvider), which is currently used by Kyma's [Istio module](https://github.com/kyma-project/istio).
+
+### Setup for Istio Access Logs
+
+To enable OpenTelemetry based access logs in Istio, execute the following steps:
+
+1. Deploy log-gateway as described above.
+
+2. Add the following envoyOtelAls extension provider to the extensionProviders list of the service mesh config by editing the `istio` ConfigMap in the `istio-system` namespace:
```
[the remark this answers is not in the document]
Comment (chrkl) [reply docs/contributor/pocs/otlp-logs.md:183]: Plural is correct.

File: docs/contributor/pocs/otlp-logs.md [hunk 4]
```diff
@@ -167,3 +167,161 @@ During the evaluation, the following potential problems and risks have been iden
...
+    1710427383130026000
+  ]
+}
+```
+
+The configured labels from the extension provider configuration result in log attributes of the OpenTelemetry log records. The log gateway enriches the log records with resource attributes to fulfill the semantic conventions. The source pod is identified by the k8sattributes processor using the client IP address.
+
+The OpenTelemetry log specification contains fields for a traceId and spanId. The extension provider does not allow to fill these fields but assign them to a log attribute. This limitation can be corrected by a transform processor in the log gateway. An [issue](https://github.com/istio/istio/issues/49911) has been created for the Istio project.
+
+The OpenTelemetry [Semantic Conventions](https://opentelemetry.io/docs/specs/semconv/) do not explicitly cover access logs yet. The shown example pick up attribute names from related areas like HTTP and URL where it made sense.
```
[the remark this answers is not in the document]
Comment (chrkl) [reply docs/contributor/pocs/otlp-logs.md]: Hope that is more clear now.

File: docs/contributor/pocs/otlp-logs.md [hunk 5]
```diff
@@ -167,3 +167,161 @@ During the evaluation, the following potential problems and risks have been iden
...
+
+### Scope and Goals
+
+Istio comes with extension providers to configure its telemetry. The [EnvoyOpenTelemetryLogProvider](https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig-ExtensionProvider-EnvoyOpenTelemetryLogProvider) allows to send access logs to an OTLP service. This part of the PoC should compare this extension provider in combination with the OpenTelemetry Collector to the [EnvoyFileAccessLogProvider](https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig-ExtensionProvider-EnvoyFileAccessLogProvider), which is currently used by Kyma's [Istio module](https://github.com/kyma-project/istio).
+
+### Setup for Istio Access Logs
+
+To enable OpenTelemetry based access logs in Istio, execute the following steps:
+
+1. Deploy log-gateway as described above.
```
[the remark this answers is not in the document]
Comment (chrkl) [reply docs/contributor/pocs/otlp-logs.md]: Linked to the right setup section. I think extracting the setup part does not make so much sense. The setup for the first and second PoC are different. The third part can re-use the second setup.

**Review by NHingerl — APPROVED** [review]

**Pull request merged**